src/Cabal2nix.hs

```diff
@@ -0,0 +1,176 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module Cabal2nix (Options (..), main, mainWithArgs) where
+
+
+import Control.Exception ( bracket )
+import Control.Lens
+import Data.Maybe ( fromMaybe )
+import Data.Monoid ( (<>) )
+import qualified Data.Set as Set
+import Data.String
+import qualified Distribution.Compat.ReadP as P
+import Distribution.Compiler
+import Distribution.Nixpkgs.Fetch
+import Distribution.Nixpkgs.Haskell
+import Distribution.Nixpkgs.Haskell.FromCabal
+import Distribution.Nixpkgs.Haskell.PackageSourceSpec
+import Distribution.Nixpkgs.Meta
+import Distribution.PackageDescription ( FlagName (..), FlagAssignment )
+import Distribution.Simple.Utils ( lowercase )
+import Distribution.System
+import Distribution.Text
+import Language.Nix
+import Options.Applicative
+import Paths_cabal2nix ( version )
+import System.Environment ( getArgs )
+import System.IO ( hFlush, stderr, stdout )
+import qualified Text.PrettyPrint.ANSI.Leijen as P2 hiding ( (<$>), (<>) )
+import Text.PrettyPrint.HughesPJClass ( Doc, Pretty (..), text, vcat, hcat, semi )
+
+
+data Options = Options
+  { optSha256                    :: Maybe String
```

Comment:
Please do not make these kind of white-space changes while committing some other (unrelated) modifications. Cosmetic changes should have their own commits (or PRs) that contain nothing but the cosmetic so that these modifications don't clutter up the diff and effectively hide the substance of the patch. The diff of this PR is quite large, but the substance of the bug fix is the removal of a *single character*. It should be obvious from the diffs, IMHO, that this is the actual change the PR makes.